lib/Dialect/FIRRTL/Import/FIRLexer.cpp

```diff
@@ -279,6 +284,38 @@ FIRToken FIRLexer::lexFileInfo(const char *tokStart) {
   }
 }
 
+/// Lex a non-standard inline Annotation file.
+///
+/// InlineAnnotation ::= '%[' (.)* ']'
+///
+FIRToken FIRLexer::lexInlineAnnotation(const char *tokStart) {
+  size_t depth = 0;
+  while (1) {
+    switch (*curPtr++) {
+    case ']':
+      if (depth == 1)
+        return formToken(FIRToken::inlineannotation, tokStart);
+      depth--;
+      break;
+    case '[':
+      depth++;
+      break;
+    case 0:
+      if (curPtr - 1 != curBuffer.end())
+        break;
+      LLVM_FALLTHROUGH;
+    case '\n': // Vertical whitespace isn't allowed in inline annotations.
```

Comment:
Do you want to eventually support escapes in these annotations?  If so, it is probably best to reject `\\` as an error for now.

Reply:
I beefed this up in a `!squash` commit. I'm handling escapes and statefully parsing `"string"` expressions now. This likely has bugs, but is better than it was.